.github/workflows/python-publish.yml

```diff
@@ -40,10 +29,33 @@ jobs:
       run: |
         python3 -m build --wheel
         python3 setup.py sdist --format=zip
-    - name: Publish package
-      env:
-        TWINE_USERNAME: __token__
-        TWINE_PASSWORD: ${{ secrets.PYPI_API_TOKEN }}
+    - name: Verify package
       run: |
         twine check dist/*
-        twine upload --repository testpypi --verbose dist/*.whl dist/*.zip
+    - name: Store package
+      uses: actions/upload-artifact@v4
+      with:
+        name: python-package-distribution
+        path: |
+          dist/*.whl
+          dist/*.zip
+        if-no-files-found: error
+
+  release:
+    runs-on: ubuntu-latest
+    needs: build
+    environment:
+      name: release
+    permissions:
+      id-token: write  # mandatory for trusted publishing
+    steps:
+    - name: Retrieve package
+      uses: actions/download-artifact@v4
+      with:
+        name: python-package-distribution
+        path: dist/
+    - name: Publish package (PyPi)
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        print-hash: true
+        repository-url: https://upload.pypi.org/legacy/  # for testing sub https://test.pypi.org/legacy/
```

Comment:
It could be recommendable to actually do both, so that if upload to TestPyPi goes wrong, the workflow fails early and PyPi stays untouched.

e.g. see https://github.com/nasa/fprime-tools/blob/eb07f23992dd2318febe932c55d0ac7890a9d3dd/.github/workflows/publish.yml#L28-L33

Reply:
I appreciate that workflow suggestion, and I will bring it up with the team in our next meeting. 

To explain, we did not publish at both the test and prod repos because experience is that `test.pypi.org` is used to get publishing working and/or to explore unreleased projects. Often, the end result of projects is to just get this working and dropped into the system. 

However, one recommended variation also publishes `develop` branch (Gitflow workflow) changes to test and then `main` branch releases to regular PyPi on either tag or commit. (In that case `.gitignore` would be set to exclude this workflow and it would be specific to those branches, or alternately a multibranch publish workflow file would be written.) In the essence of simplicity and straightforward application this was left to advanced users to factor.